src/internal/form.ts

```diff
@@ -357,10 +357,11 @@ export class FormControlController implements ReactiveController {
    * event will be cancelled before being dispatched.
    */
   emitInvalidEvent(originalInvalidEvent?: Event) {
-    const slInvalidEvent = new CustomEvent<void>('sl-invalid', {
+    const slInvalidEvent = new CustomEvent<Record<PropertyKey, never>>('sl-invalid', {
       bubbles: false,
       composed: false,
-      cancelable: true
+      cancelable: true,
+      detail: {}
```

Comment:
Changed `null` details to an empty object to be consistent with all other events in Shoelace

Reply:
Thanks a lot :+1:
May I ask: What's the advantage of using
`type SlInvalidEvent = CustomEvent<Record<PropertyKey, never>>;` instead of just using `type SlInvalidEvent = CustomEvent<{}>;` (see [here](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAbwL4CgUwJ5mHAygGwEkA7ANwEN8BLAEwFFThj4BeOAYQFcBnGCAWwZMYAHgBKwAMbQaIgApQI2WBgDSwDABo4xYIygA+AwG50WHARIVq9RswBMcNl14ChzEchNoaU-OSgcAHN8CAAjSkQUODgqZmAoADNySRwAcVCI-HcYAAlyYhp8BO4cgFlyMCiYmIBybnwAWjjrWlqALjwiMkpaHNMauHqmlt6aew6uqzGc+wG4VFQUGghJTn5hADpyGlthABkqXiYEgAph5p6bWu1TvQBKJwNquGlibghizdCgu9JN3wwchUfCbGAACyOAH0ANYaKHQ4gQGBQ8gUEHkMLFKFQjDAFFIqEJRRQe6mJBknyrdZbHZ7ZiHY66KDnBqXVrjG5wP6PFjPBDRV4Qd6fYDfCC-PQA-HA0EQ6FwjAI7hQpEotGyzHYqHFHYqvhEqAkykU4xAA))?

Reply:
Supprisingly, `{}` doesn't actually mean empty object when used as a type in Typescript. I actually initially used `CustomEvent<{}>` but then got a compiler warning about it not doing what you expect.